bears/general/KeywordBear.py

```diff
@@ -2,18 +2,71 @@
 
 from coalib.bearlib import deprecate_settings
 from coalib.bears.LocalBear import LocalBear
+from coalib.results.Diff import Diff
 from coalib.results.Result import RESULT_SEVERITY, Result
 
+from bears.general.AnnotationBear import AnnotationBear
+from coalib.results.SourceRange import SourceRange
+
+
+def _get_comments(dependency_results):
+    if not dependency_results:
+        return {}
+
+    annotation_bear_results = dependency_results.get('AnnotationBear')
+    if (not annotation_bear_results or
+            not isinstance(annotation_bear_results, list)):
+        return {}
+
+    comments = set()
+    for result in annotation_bear_results:
+        comments = comments.union(result.contents.get('comments', []))
+    return comments
+
+
+def generate_diff(dependency_results, file, filename,
+                  line, line_number, pos):
+    comments = _get_comments(dependency_results)
```

Comment:
you can also spot those performance thingies because a function that gennerates a diff shouldn't be responsible for getting te information it needs for it as well